geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/KryoBufferSimpleFeature.scala

```diff
@@ -42,8 +42,9 @@ class KryoBufferSimpleFeature(sft: SimpleFeatureType,
   private var userDataOffset: Int = -1
 
   private var binaryTransform: () => Array[Byte] = input.getBuffer
+  private var reserializeTransform: () => Array[Byte] = input.getBuffer
 
-  def transform(): Array[Byte] = binaryTransform()
+  def transform(): Array[Byte] = if (offsets.contains(-1)) reserializeTransform() else binaryTransform()
```

Comment:
this method exposes the private vars above, without allowing them to be modified.